AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift

```diff
@@ -177,14 +179,19 @@ class AWSDeviceTracker: NSObject, CLLocationManagerDelegate, AWSDeviceTrackingBe
                                                                             deviceID: deviceID) })
             try self.locationStore.insert(positions: positionsToStore)
         } catch {
-            // TODO: send error on Hub
+            let geoError = Geo.Error.unknown(
+                GeoPluginErrorConstants.errorSavingLocationsToLocalStore.errorDescription, GeoPluginErrorConstants.errorSavingLocationsToLocalStore.recoverySuggestion,
+                error)
+            sendHubErrorEvent(error: geoError, locations: receivedLocations)
         }
     }
 
     func mapReceivedLocationsToPositions(receivedLocations: [CLLocation], currentTime: Date) -> [Position] {
         guard let deviceId = UserDefaults.standard.object(forKey: AWSDeviceTracker.deviceIDKey) as? String else {
             Amplify.log.error("Not able to fetch deviceId from UserDefaults")
-            // TODO: send error on hub
+            let error = Geo.Error.unknown(
```

Comment:
should we add a new error constant for this scenario?

Reply:
I added a new `Geo.Error.internalPluginError`